DatadogInternal/Sources/Codable/AnyDecoder.swift

```diff
@@ -856,3 +856,26 @@ private extension DecodingError {
         return .typeMismatch(type, context)
     }
 }
+
+extension SingleValueDecodingContainer {
+    /// Decodes a passthrough value of the given type.
+    ///
+    /// This method can succeed only when using `AnyDecoder`.
+    ///
+    /// - returns: A passthrough value if any.
+    /// - throws: `DecodingError.typeMismatch` if the value was not passthrough.
+    func passthrough() throws -> PassthroughAnyCodable {
+        guard let container = self as? _AnyDecoder.SingleValueContainer else {
+            throw DecodingError.typeMismatch(PassthroughAnyCodable.self, DecodingError.Context(
+                codingPath: codingPath,
+                debugDescription: "Single Value Container cannot decode passthrough value."
+            ))
+        }
+
+        guard let value = container.value as? PassthroughAnyCodable else {
+            throw DecodingError.typeMismatch(PassthroughAnyCodable.self, in: container)
+        }
+
+        return value
+    }
```

Comment:
Any errors thrown from here are ignored in:
```swift
} else if let passthrough = try? container.passthrough() {
```
so I don't see exactly how it works 🤔 💭. It looks that this can eventually be:
```swift
func passthrough() -> PassthroughAnyCodable? {
   return (container as? _AnyDecoder.SingleValueContainer)?.value as? PassthroughAnyCodable
}
```

Reply:
It could return `nil` indeed, but error escalation is a best and follow convention in `Codable`.

Reply:
Thing is, errors are ignored in caller code, so this is effectively a boilerplate. The way we use Swift's structured error handling here is not idiomatic. We throw error only to receive `nil` in `try?`, which is very confusing and makes an illusion of more control over this piece than we actually have (e.g. we customise error metadata, but other code ignores it). Tests we added only skew the real code coverage.

Not a blocker, but I'd suggest to keep it simple if we can't make use of error handling here.

Reply:
The fact that it's ignored a caller doesn't mean we shouldn't escalate errors. But I get your point, it's more valuable to throw something if we have a signature like:
```swift
func passthrough<T>(_ type T.Type) throws -> T where T: PassthroughAnyCodable
```
Here, it doesn't really make sense! I can remove it